src/ingestors/rarible/index.ts

```diff
@@ -0,0 +1,100 @@
+        if (new URL(url).hostname !== 'www.rarible.com' && new URL(url).hostname !== 'rarible.com') {
+            return false;
+        }
+             
+        const { chainId,  contractAddress } = await await raribleOnchainDataFromUrl(url)
+        return !!chainId && !!contractAddress
+    }   
+       
+    async supportsContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<boolean> {
+        const { chainId, contractAddress } = contract;
+        if (!chainId || !contractAddress) {
+          return false;
+        }
+        const url = await raribleUrlForValidBaseEthCollection(chainId, contractAddress, resources.alchemy);
+        return !!url;
+    }
+    
+
+    async createMintTemplateForUrl(resources: MintIngestorResources, url: string): Promise<MintTemplate> {
+        const isCompatible = await this.supportsUrl(resources, url);
+        if (!isCompatible) {
+          throw new MintIngestorError(MintIngestionErrorName.IncompatibleUrl, 'Incompatible URL');
+        }
+
+        const { chainId, contractAddress } = await raribleOnchainDataFromUrl(url);
+        if (!chainId || !contractAddress) {
+          throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+        }
+
+        return this.createMintForContract(resources, { chainId, contractAddress, url });
+    }
+   
+    async createMintForContract(resources: MintIngestorResources, contract: MintContractOptions): Promise<MintTemplate> {
+        const { chainId, contractAddress } = contract;
+        if (!chainId || !contractAddress) {
+            throw new MintIngestorError(MintIngestionErrorName.MissingRequiredData, 'Missing required data');
+        }
+        
+        const isValid = await raribleUrlForValidBaseEthCollection(chainId, contractAddress, resources.alchemy)
+        if (!isValid) {
+          throw new MintIngestorError(MintIngestionErrorName.CouldNotResolveMint, 'Project not found');
+        }
+        
+        const mintBuilder = new MintTemplateBuilder()
+            .setMintInstructionType(MintInstructionType.EVM_MINT)
+            .setPartnerName('Rarible');
+
+        if (contract.url) {
+            mintBuilder.setMarketingUrl(contract.url);
+        }
+       const { 
+            name, 
+            description, 
+            imageURI, 
+            unitMintPrice, 
+            unitMintTokenAddress, 
+            unitPerWallet, 
+            conditionProof, 
+            ownerAddress,
+            startDate,
+            stopDate
+        } = await raribleContractMetadataGetter(chainId, contractAddress, resources.alchemy)
+        
+        mintBuilder.setName(name).setDescription(description).setFeaturedImageUrl(imageURI);
+        mintBuilder.setMintOutputContract({ chainId: chainId, address: contractAddress });
+        mintBuilder.setCreator({
+                name: 'not available',
```

Comment:
This will show up in the UI which we don't want.

It does look like rarible.org don't include this in the public API with key, but rarible.com do provide it in their API which appears accessible

```
curl 'https://rarible.com/marketplace/api/v4/profiles/list' \
  -H 'accept: */*' \
  -H 'content-type: application/json' \
  --data-raw '["0xbab964a06fbd884b0ad0efccb86f5931248858cc"]'
```

This has the name and other details which would make this much better! 🙂 

![CleanShot 2024-07-23 at 09 57 21@2x](https://github.com/user-attachments/assets/12ece8f8-7bb6-42f1-be9b-8dfc5c34ad7a)

Reply:
Hey, thanks. Been in and out of their telegram chat and couldn't get this endpoint.